rmw_opendds_cpp/src/type_support_common.hpp

```diff
@@ -27,71 +27,56 @@
-  } \
-  const rosidl_service_type_support_t * TYPE_SUPPORT = \
-    get_service_typesupport_handle( \
-    TYPE_SUPPORTS, rosidl_typesupport_opendds_c__identifier); \
-  if (!TYPE_SUPPORT) { \
-    TYPE_SUPPORT = get_service_typesupport_handle( \
-      TYPE_SUPPORTS, rosidl_typesupport_opendds_cpp::typesupport_identifier); \
-    if (!TYPE_SUPPORT) { \
-      char __msg[1024]; \
-      snprintf( \
-        __msg, 1024, \
-        "type support handle implementation '%s' (%p) does not match valid type supports " \
-        "('%s' (%p), '%s' (%p))", \
-        TYPE_SUPPORTS->typesupport_identifier, \
-        static_cast<const void *>(TYPE_SUPPORTS->typesupport_identifier), \
-        rosidl_typesupport_opendds_cpp::typesupport_identifier, \
-        static_cast<const void *>(rosidl_typesupport_opendds_cpp::typesupport_identifier), \
-        rosidl_typesupport_opendds_c__identifier, \
-        static_cast<const void *>(rosidl_typesupport_opendds_c__identifier)); \
-      RMW_SET_ERROR_MSG(__msg); \
-      return RET_VAL; \
-    } \
+  const rosidl_message_type_support_t * ts = get_message_typesupport_handle(
+  //type_support, rosidl_typesupport_opendds_c__identifier);
+    type_support, "rosidl_typesupport_c"); //?? temp
+  if (!ts) {
+//  ts = get_message_typesupport_handle(type_support, rosidl_typesupport_opendds_cpp::typesupport_identifier);
+    ts = get_message_typesupport_handle(type_support, "rosidl_typesupport_cpp"); //?? temp
+    if (!ts) {
+      RMW_SET_ERROR_MSG_WITH_FORMAT_STRING("type support implementation '%s' does not match '%s'",
+        type_support->typesupport_identifier, rosidl_typesupport_opendds_cpp::typesupport_identifier);
+    }
   }
+  return ts;
+}
 
+inline const rosidl_service_type_support_t *
+rmw_get_service_type_support(const rosidl_service_type_support_t * type_support)
+{
+  if (!type_support) {
+    RMW_SET_ERROR_MSG("type support is null");
+    return nullptr;
+  }
+  const rosidl_service_type_support_t * ts = get_service_typesupport_handle(
+//  type_support, rosidl_typesupport_opendds_c__identifier);
+    type_support, "rosidl_typesupport_cpp"); //?? temp
+  if (!ts) {
+    ts = get_service_typesupport_handle(type_support, rosidl_typesupport_opendds_cpp::typesupport_identifier);
+    if (!ts) {
+      RMW_SET_ERROR_MSG_WITH_FORMAT_STRING("type support implementation '%s' does not match '%s'",
+        type_support->typesupport_identifier, rosidl_typesupport_opendds_cpp::typesupport_identifier);
+    }
+  }
+  return ts;
+}
 
 inline std::string
 _create_type_name(
   const message_type_support_callbacks_t * callbacks,
   const std::string & sep)
 {
-  return
-    std::string(callbacks->message_namespace) +
-    "::" + sep + "::dds_::" + callbacks->message_name + "_";
+/*
+  return std::string(callbacks->message_namespace) + "::" + sep + "::dds_::" + callbacks->message_name + "_";
+*/
+  return std::string("callbacks->message_namespace::" + sep + "::dds_::callbacks->message_name_"); //?? temp
```

Comment:
This change may get us to a running state sooner which is the goal of this PR. Are `message_namespace` and `message_name` null at this point?